Build fails since `bun@1.0.17`

After fighting a bit to get this repo to work, we figured out that with recent `bun` versions it simply doesn't build (and also doesn't install dependencies because `bun run build` is a `prepare` pre-script).

```
➜  evmgateway git:(main) ✗ bun install                       
bun install v1.0.17 (5e60861c)
  ⚙️  @ensdomains/op-gateway [1/5] $ echo 'building l1-gateway...' && bun run clean && bun run build:cjs && bun run build:esm && bun run build:types
building l1-gateway...
$ rm -fr _cjs _esm _types
$ tsc --project tsconfig.build.json --module commonjs --outDir ./_cjs --removeComments --verbatimModuleSyntax false && echo > ./_cjs/package.json '{"type":"commonjs"}'
src/index.ts(1,28): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.
src/L1ProofService.ts(8,52): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.
src/server.ts(3,28): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.
src/worker.ts(2,29): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.
src/worker.ts(12,38): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.

error: prepare script from "@ensdomains/l1-gateway" exited with code 2 (SIGINT)
```

We also identified that this is related to the [parallelization of lifecycle scripts since v1.0.17](https://bun.sh/blog/bun-v1.0.17#lifecycle-scripts-now-run-in-parallel). There's an easy fix to it which I'll open a PR for shortly. 🫡

--- a/arb-gateway/README.md
+++ b/arb-gateway/README.md
@@ -9,25 +9,24 @@ To get started, you need to have an RPC URL for both Ethereum Mainnet and Arbitr
 ## How to use arb-gateway locally via cloudflare dev env (aka wrangler)
 
 ```
-npm install -g bun
 cd arb-gateway
+npm install -g bun wrangler
 bun install
 touch .dev.vars
 ## set L1_PROVIDER_URL, L2_PROVIDER_URL, L2_ROLLUP
-yarn dev
+bun run dev
 ```
 
 ## How to deploy arb-gateway to cloudflare
 
 ```
 cd arb-gateway
-npm install -g wrangler
-wrngler login
-
+npm install -g bun wrangler
+wrangler login
 wrangler secret put L1_PROVIDER_URL
 wrangler secret put L2_PROVIDER_URL
 wrangler secret put L2_ROLLUP
-yarn deploy
+bun run deploy
 ```
 
 ## How to test
@@ -39,6 +38,5 @@ yarn deploy
 5. Navigate to the Gateway directory using `cd ./arb-gateway`.
 6. Start the Gateway by running `bun run start -u http://127.0.0.1:8545/ -v http://127.0.0.1:8547/ -p 8089`.
 7. Open another Terminal Tab and navigate to the verifier directory using `cd ./arb-verifier/`.
-8. Deploy contracts to the node using the command ` npx hardhat --network arbDevnetL2 deploy && npx hardhat --network arbDevnetL1 deploy `.
+8. Deploy contracts to the node using the command `npx hardhat --network arbDevnetL2 deploy && npx hardhat --network arbDevnetL1 deploy`.
 9. Run the test using the command `bun run test`.
-

--- a/arb-gateway/wrangler.toml
+++ b/arb-gateway/wrangler.toml
@@ -8,6 +8,6 @@ compatibility_date = "2023-10-13"
 port = 8080
 
 [build]
-command = "yarn build"
+command = "bun run build"
 
 [vars]

--- a/bunfig.toml
+++ b/bunfig.toml
@@ -1,0 +1,2 @@
+# Only run 1 script at a time to prevent race conditions (see https://github.com/ensdomains/evmgateway/issues/31) 
+install.concurrentScripts = 1

--- a/l1-gateway/README.md
+++ b/l1-gateway/README.md
@@ -12,29 +12,29 @@ or historic values for storage variables of any contract.
 ## How to use l1-gateway locally via cloudflare dev env (aka wrangler)
 
 ```
-npm install -g bun
 cd l1-gateway
+npm install -g bun wrangler
 bun install
 touch .dev.vars
 ## set WORKER_PROVIDER_URL
-yarn dev
+bun run dev
 ```
 
 ## How to deploy l1-gateway to cloudflare
 
 ```
 cd l1-gateway
-npm install -g wrangler
-wrngler login
+npm install -g bun wrangler
+wrangler login
 wrangler secret put WORKER_PROVIDER_URL
-yarn deploy
+bun run deploy
 ```
 
 ## How to test
 
 ```
 cd ../l1-verifier l1-gateway
-PROVIDER_URL=$PROVIDER_URL TARGET_ADDRESS=$TARGET_ADDRESS yarn remote_test
+PROVIDER_URL=$PROVIDER_URL TARGET_ADDRESS=$TARGET_ADDRESS bun run remote_test
 ```
 
 ## Current deployments

--- a/l1-gateway/wrangler.toml
+++ b/l1-gateway/wrangler.toml
@@ -8,6 +8,6 @@ compatibility_date = "2023-10-13"
 port = 8080
 
 [build]
-command = "yarn build"
+command = "bun run build"
 
 [vars]

--- a/op-gateway/README.md
+++ b/op-gateway/README.md
@@ -7,32 +7,32 @@ For a detailed readme and usage instructions, see the [monorepo readme](https://
 ## How to use op-gateway locally via cloudflare dev env (aka wrangler)
 
 ```
-npm install -g bun
 cd op-gateway
+npm install -g bun wrangler
 bun install
 touch .dev.vars
 ## set L1_PROVIDER_URL, L2_PROVIDER_URL, L2_OUTPUT_ORACLE, DELAY=5
-yarn dev
+bun run dev
 ```
 
 ## How to deploy op-gateway to cloudflare
 
 ```
 cd op-gateway
-npm install -g wrangler
-wrngler login
+npm install -g bun wrangler
+wrangler login
 wrangler secret put L1_PROVIDER_URL
 wrangler secret put L2_PROVIDER_URL
 wrangler secret put L2_OUTPUT_ORACLE
 wrangler secret put DELAY
-yarn deploy
+bun run deploy
 ```
 
 ## How to test
 
 ```
 cd ../l1-verifier l1-gateway
-PROVIDER_URL=$PROVIDER_URL TARGET_ADDRESS=$TARGET_ADDRESS yarn remote_test
+PROVIDER_URL=$PROVIDER_URL TARGET_ADDRESS=$TARGET_ADDRESS bun run remote_test
 ```
 
 ## Current deployments

--- a/op-gateway/wrangler.toml
+++ b/op-gateway/wrangler.toml
@@ -8,6 +8,6 @@ compatibility_date = "2023-10-13"
 port = 8080
 
 [build]
-command = "yarn build"
+command = "bun run build"
 
 [vars]

